### ISSUE TITLE
Blake2 method `KeyInit::new` is incorrect and not tested

### ISSUE THREAD

**Issue by vlad9486** [RustCrypto/hashes#433]
Now it is crashing:

```
#[cfg(test)]
#[test]
fn blake2b_trivial() {
    use blake2::{Blake2bMac512, digest::KeyInit};

    <Blake2bMac512 as KeyInit>::new(&Default::default());
}
```

Because of this
https://github.com/RustCrypto/hashes/blob/af0567f2605901bf6b4811775427c8f1dd7e350d/blake2/src/macros.rs#L328

This line is passing key as salt, but it should not.

**Comment by vlad9486** [comment]
#432

**Comment by newpavlov** [comment]
The issue is fixed in v0.10.5. I will close this issue after tests for `KeyInit::new` will be added.
